File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& arr) {
        
        
        vector<vector<int>> ans;
        
        if(arr.size()==0){
            return ans;
        }
        
        sort(arr.begin(),arr.end());
        
        vector<int> temp = arr[0];
        
        for(auto it:arr){
            if(it[0]<=temp[1]){
                temp[1] = max(it[1],temp[1]);
            }else{
                ans.push_back(temp);
                temp = it;
            }
        }
        
        ans.push_back(temp);
        return ans;
        
        
        
        
        
        
        
    }
};

File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, arr: List[List[int]]) -> List[List[int]]:
        
        arr.sort()
        
        ans = []
        
        temp = arr[0]
        
        for it in arr:
            
            if it[0]<=temp[1]:
                temp[1] = max(it[1],temp[1])
            else:
                ans.append(temp)
                temp = it
        
        ans.append(temp)
        
        return ans
    